api/grpc/controller_test.go

```diff
@@ -0,0 +1,263 @@
+// Copyright 2021 Alvalor S.A.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package grpc
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	tmock "github.com/stretchr/testify/mock"
+
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/model/flow"
+
+	"github.com/optakt/flow-dps/models/dps/mock"
+)
+
+func TestNewController(t *testing.T) {
+	c := NewController(nil)
+	assert.NotNil(t, c)
+}
+
+func TestController_GetRegister(t *testing.T) {
+	var (
+		testHeight uint64 = 128
+		lastHeight uint64 = 256
+
+		testKey   = []byte(`testKey`)
+		testValue = []byte(`testValue`)
+	)
+
+	tests := []struct {
+		desc string
+
+		reqHeight *uint64
+		reqKey    []byte
+
+		mockValue []byte
+		mockErr   error
+
+		wantHeight uint64
+		wantResp   *GetRegisterResponse
+		wantErr    assert.ErrorAssertionFunc
+	}{
+		{
+			desc: "nominal case, height given",
+
+			reqHeight: &testHeight,
+			reqKey:    testKey,
+
+			mockValue: testValue,
+
+			wantHeight: testHeight,
+			wantResp: &GetRegisterResponse{
+				Height: testHeight,
+				Key:    testKey,
+				Value:  testValue,
+			},
+			wantErr: assert.NoError,
+		},
+		{
+			desc: "nominal case, no height given",
+
+			reqKey: testKey,
+
+			mockValue: testValue,
```

Comment:
Wouldn't it make more sense to call these `haveXXX`? In analogy to `wantXXX`?

Reply:
If you find it nicer, sure. For me what's nice with prefixing with `mock` is that it's clear that it's something returned by the mock, whereas with `have` it could be mistaken for a func parameter. WDYT?

Reply:
Well, generally, test vectors qualify things as inputs to the test and outputs from the test, and generally, that's shortened to `have` and `want`. I don't think it's relevant whether an input comes from a programmed mock or not :D

I leave it up to you.

Reply:
I find it clearer this way, so if you don't mind let's keep it this way for now 👍